cmd/es-rollover/app/init/flags.go

```diff
@@ -16,48 +16,82 @@ package init
 
 import (
 	"flag"
+	"log"
 
 	"github.com/spf13/viper"
 
 	"github.com/jaegertracing/jaeger/cmd/es-rollover/app"
+	cfg "github.com/jaegertracing/jaeger/pkg/es/config"
 )
 
 const (
-	shards                       = "shards"
-	replicas                     = "replicas"
-	prioritySpanTemplate         = "priority-span-template"
-	priorityServiceTemplate      = "priority-service-template"
-	priorityDependenciesTemplate = "priority-dependencies-template"
-	prioritySamplingTemplate     = "priority-sampling-template"
+	shards                  = "shards"
+	replicas                = "replicas"
+	prioritySpanTemplate    = "priority-span-template"
+	priorityServiceTemplate = "priority-service-template"
 )
 
 // Config holds configuration for index cleaner binary.
 type Config struct {
 	app.Config
-	Shards                       int
-	Replicas                     int
-	PrioritySpanTemplate         int
-	PriorityServiceTemplate      int
-	PriorityDependenciesTemplate int
-	PrioritySamplingTemplate     int
+	cfg.Indices
 }
 
 // AddFlags adds flags for TLS to the FlagSet.
 func (*Config) AddFlags(flags *flag.FlagSet) {
-	flags.Int(shards, 5, "Number of shards")
-	flags.Int(replicas, 1, "Number of replicas")
-	flags.Int(prioritySpanTemplate, 0, "Priority of jaeger-span index template (ESv8 only)")
-	flags.Int(priorityServiceTemplate, 0, "Priority of jaeger-service index template (ESv8 only)")
-	flags.Int(priorityDependenciesTemplate, 0, "Priority of jaeger-dependencies index template (ESv8 only)")
-	flags.Int(prioritySamplingTemplate, 0, "Priority of jaeger-sampling index template (ESv8 only)")
+	flags.Int(shards, 5, "(deprecated, will be replaced with num-shards-span,num-shards-service,num-shards-sampling,num-shards-dependencies, if .num-shards set, it will be used as global settings for span,service,sampling,dependencies index) Number of shards")
```

Comment:
sorry but I really don't want to be spending time reviewing unnecessary changes. We are not planning to deprecate these flags - we are planning to deprecate the whole binaries once Jaeger-v2 is released. In case you misunderstood my suggestion, I'll repeat:
  * undo all changes to the flags
  * the only changes to flags parsing that is needed is instead of `c.Shards = v.GetInt(shards)` we can do
```
c.Indices.Spans.TemplateNumShards = v.GetInt(shards)
c.Indices.Services.TemplateNumShards = v.GetInt(shards)
c.Indices.Dependencies.TemplateNumShards = v.GetInt(shards)
c.Indices.Sampling.TemplateNumShards = v.GetInt(shards)
```

Reply:
Oh, sorry, I misunderstood. Thank you for your patient explanation. I will update it according to your suggestion